engine/src/main/java/org/pentaho/di/www/GetDatabaseDetailsServlet.java

```diff
@@ -0,0 +1,111 @@
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.www;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.pentaho.di.core.database.DatabaseInterface;
+import org.pentaho.di.core.database.DatabaseMeta;
+import org.pentaho.di.core.exception.KettlePluginException;
+import org.pentaho.di.core.plugins.DatabasePluginType;
+import org.pentaho.di.core.plugins.PluginInterface;
+import org.pentaho.di.core.plugins.PluginRegistry;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+
+/**
+ *
+ *This class contains api's to fetch registered databases.
+ *
+ * @since 2024
+ * @author sVarma1992
+ *
+ */
+public class GetDatabaseDetailsServlet extends BaseHttpServlet implements CartePluginInterface {
+
+  private static Class<?> PKG = GetDatabaseDetailsServlet.class; // for i18n purposes, needed by Translator2!!
+
+  private static final long serialVersionUID = 3634806745372015720L;
+
+  public static final String CONTEXT_PATH = "/kettle/registeredDatabasePlugins";
+
+  public GetDatabaseDetailsServlet() {
+  }
+
+  public void doGet( HttpServletRequest request, HttpServletResponse response ) throws ServletException,
+          IOException {
+
+    if ( isJettyMode() && !request.getContextPath().startsWith( CONTEXT_PATH ) ) {
+      return;
+    }
+
+    response.setStatus( HttpServletResponse.SC_OK );
+    response.setCharacterEncoding( "UTF-8" );
+    response.setContentType( "application/json;charset=UTF-8" );
+    JSONArray previewJson = new JSONArray();
+
+    PrintWriter out = response.getWriter();
+    PluginRegistry registry = PluginRegistry.getInstance();
+    List<PluginInterface> stepPlugins = registry.getPlugins( DatabasePluginType.class );
+
+    for ( PluginInterface pluginInterface : stepPlugins ) {
+      if ( pluginInterface.getIds().length > 0 ) {
+        JSONObject dbJson = new JSONObject();
+        dbJson.put( "id", pluginInterface.getIds()[0] );
+        dbJson.put( "name", pluginInterface.getName() );
+        try {
+          DatabaseInterface db = (DatabaseInterface) registry.loadClass( pluginInterface );
+          JSONArray accessTypes = new JSONArray();
+          for ( int id : db.getAccessTypeList() ) {
+            accessTypes.add( DatabaseMeta.getAccessTypeDesc( id ) );
+          }
+          dbJson.put( "accessTypes", accessTypes );
+        } catch ( KettlePluginException e ) {
+          throw new ServletException( e.getMessage() );
+        }
+        previewJson.add( dbJson );
+      }
+    }
+
+    JSONObject finalJsonOutput = new JSONObject();
+    finalJsonOutput.put( "plugins", previewJson );
+    out.print( finalJsonOutput );
+  }
+
+  public String toString() {
+    return "Registered Plugins Handler";
```

Comment:
can this be a static member and made constant

Reply:
This was also created similar to the toString() method in ExecuteTransServlet Vamsi.

Reply:
Since, we are writing new code and this seems to be a repetitive String in multiple new Servlets suggested to make it as constant.
we have a choice to make it better now :)